include/fbgemm/Utils.h

```diff
@@ -162,6 +162,33 @@ struct FBGEMM_API BlockingFactors {
   int NCB;
 };
 
+/**
+ * @brief A struct to represent the partition information for the threads on the
+ * m and n dimensions.
+ */
+struct FBGEMM_API thread_type_t {
+  int m_num_threads;
+  int n_num_threads;
+  int m_thread_id;
+  int n_thread_id;
+
+  std::string toString() const {
+    std::string out = "";
+    out += "m num threads: " + std::to_string(m_num_threads) + ", ";
+    out += "n num threads: " + std::to_string(n_num_threads) + ", ";
+    out += "m thread id: " + std::to_string(m_thread_id) + ", ";
+    out += "n thread id: " + std::to_string(n_thread_id);
+    return out;
+  }
+};
+
+/**
+ * @brief A heuristic way to partition the threads across m and n dimensions for
+ * parallelization.
+ */
+FBGEMM_API thread_type_t
+fbgemmGetThreadPartition(int m, int n, int num_threads, int thread_id);
```

Comment:
We may want to pass n_align argument (default value to be 64) so each n partition is cache line aligned.

Reply:
Added n_align as one additional parameter. When we also partition along the n dimension and align the cache line (64), we need to do more careful design for the thread partition heuristics. Didn't change the code yet...

Reply:
Changed the code and added the handler for the alignment of cache line and avoid the false sharing due to the wrong cache line border case.